common/src/main/kotlin/org/neo4j/connectors/kafka/data/PropertyConstraints.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [https://neo4j.com]
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.connectors.kafka.data
+
+import org.neo4j.driver.Session
+
+enum class ConstraintEntityType(val value: String) {
+  NODE("NODE"),
+  RELATIONSHIP("RELATIONSHIP"),
+}
+
+enum class ConstraintType(val value: String) {
+  NODE_KEY("NODE_KEY"),
+  NODE_UNIQUENESS("UNIQUENESS"),
+  NODE_EXISTENCE("NODE_PROPERTY_EXISTENCE"),
+  RELATIONSHIP_KEY("RELATIONSHIP_KEY"),
+  RELATIONSHIP_UNIQUENESS("UNIQUENESS"),
+  RELATIONSHIP_EXISTENCE("RELATIONSHIP_PROPERTY_EXISTENCE")
+}
+
+val NODE_KEY_CONSTRAINTS =
+    listOf(
+        ConstraintType.NODE_KEY.value,
+        ConstraintType.NODE_UNIQUENESS.value,
+        ConstraintType.NODE_EXISTENCE.value)
+
+val RELATIONSHIP_KEY_CONSTRAINTS =
+    listOf(
+        ConstraintType.RELATIONSHIP_KEY.value,
+        ConstraintType.RELATIONSHIP_UNIQUENESS.value,
+        ConstraintType.RELATIONSHIP_EXISTENCE.value)
+
+data class ConstraintData(
+    val entityType: String,
+    val constraintType: String,
+    val labelOrType: String,
+    val properties: List<String>
+)
+
+fun fetchConstraintData(session: Session): List<ConstraintData> {
+  return try {
+    session.run("SHOW CONSTRAINTS").list().map {
+      ConstraintData(
+          entityType = it.get("entityType").asString(),
+          constraintType = it.get("type").asString(),
+          labelOrType = it.get("labelOrTypes").asList()[0].toString(),
```

Comment:
my first intention was to add integration test but I couldn't wrap my head around how exactly an integration test to this case should be and how to read logs, I think, to be able to test this case we need to add different integration test setup rather than what we have today, or any suggestions on that?

Reply:
I'm surprised the existing integration tests did not catch the error actually. It should fail hard since `labelOrType` is not part of the result column names... unless `toList` converts `null` to an empty list? 🤔 
You don't necessarily need an integration test just for this, but it would be nice to understand why this did not cause a hard failure.

Reply:
I figured out why integration tests are not failing, because in the beginning there is no constraints when we build up the sink connector for the tests, so query result is an empty list.

Reply:
and this will also cause getting warning log messages for node and relationship pattern integration tests.